### PR TITLE
#110 comments addressed

### PR DESCRIPTION
**Pull request by ilijapuaca**

### CODE REVIEW

File: cpp/modules/luma.gl/garrow/src/array.cc [hunk 1]
```diff
@@ -44,36 +44,32 @@ void Array::setData(const std::shared_ptr<arrow::Array>& data, wgpu::BufferUsage
...
+  // If child_data isn't empty, data is a list array
+  if (!data->data()->child_data.empty()) {
+    uint64_t offset = 0;
 
-  uint64_t offset = 0;
-  auto buffers = data->data()->buffers;
-  // Starting from buffer at index 1, as the first buffer is used for null bitmap
-  for (int i = 1; i < buffers.size(); i++) {
-    auto buffer = buffers[i];
+    // Go through child data for list data types and use child buffers
```
Comment (ibgreen) [inline cpp/modules/luma.gl/garrow/src/array.cc:60]: I got lost when I read this comment. I can understand that we are iterating through "chunks" of a data array, but that is different from "children" in my understanding. 

Children is related to the hierarchy of column types, (a list having a child being the primitive element of the type, a struct having multiple children etc).

Be extra careful distinguishing between chunks and children, and maybe make an short "ASCII diagram" in comments above explain what children contains in the list case?

File: cpp/modules/deck.gl/layers/src/line-layer/line-layer.cpp [hunk 2]
```diff
@@ -172,6 +171,8 @@ auto LineLayer::_getModel(wgpu::Device device) -> std::shared_ptr<lumagl::Model>
   auto instancedAttributes = this->attributeManager->update(this->props()->data);
   model->setInstancedAttributes(instancedAttributes);
 
+  model->vertexCount = static_cast<int>(positionData.size());
```
[the remark this answers is not in the document]
Comment (ilijapuaca) [reply cpp/modules/deck.gl/layers/src/line-layer/line-layer.cpp:174]: There is already something called `instanceCount` in `Model::draw`. Current naming is in line with wgpu Draw signature:
`void Draw(uint32_t vertexCount, uint32_t instanceCount = 1, uint32_t firstVertex = 0, uint32_t firstInstance = 0) const;`

File: cpp/modules/luma.gl/garrow/src/array.cc [hunk 3]
```diff
@@ -44,36 +44,32 @@ void Array::setData(const std::shared_ptr<arrow::Array>& data, wgpu::BufferUsage
...
+    this->_buffer = this->_createBuffer(this->_device, bufferSize, usage);
   }
 
   // TODO(ilija@unfolded.ai): Handle arrays with null values correctly
   if (data->null_count() > 0) {
     throw new std::runtime_error("Data with null values is currently not supported");
   }
 
-  // TODO(ilija@unfolded.ai): Not sure if this is the best way to extract the data out of any data type
+  // If child_data isn't empty, data is a list array
```
[the remark this answers is not in the document]
Comment (ilijapuaca) [reply cpp/modules/luma.gl/garrow/src/array.cc:56]: Is there an `ArrayData` type that handles structs?

There are many subtle things that may need special treatment to make this work across all the types, as `Array`/`ArrayData` types are very complex. I hope to get some time to go over these in detail and consolidate data inspection/extraction

File: cpp/modules/luma.gl/garrow/src/array.cc (hunk 1 above)
Comment (ilijapuaca) [reply cpp/modules/luma.gl/garrow/src/array.cc:60]: There are no chunks in this use case. There's a separate type called `ChunkedArray` that a table contains, we're working with non-chunked `Array`s here which are result of accessor processing.

We're dealing with children here, which as you noted, covers the case of list arrays. We iterate over each of the arrays in the list and extract its data

Comment (ibgreen) [reply cpp/modules/luma.gl/garrow/src/array.cc:60]: I think what gets me here is that my mental image is that a list array only has one child, so this iteration is confusing - child is a very generic term and I wish arrow had used a more specific word.

**Pull request merged**